src/com/google/enterprise/adaptor/fs/FileDelegate.java

```diff
@@ -136,15 +135,15 @@
   Path resolveDfsLink(Path doc) throws IOException;
 
   /**
-   * Returns a list of DFS links contained within a DFS namespace.
-   * The supplied path would typically be like
+   * Returns a {@link DirectoryStream} of DFS links contained within a
+   * DFS namespace. The supplied path would typically be like
    * {@code \\\\server\\namespace} or {@code \\\\domain\\namespace}.
    *
    * @param doc the DFS UNC path to a DFS namespace
-   * @returns a List of DFS link paths
+   * @returns a DirectoryStream of DFS link paths
    * @throws IOException if doc is not a DFS namespace or is not accessable
    */
-  List<Path> enumerateDfsLinks(Path doc) throws IOException;
+  DirectoryStream<Path> newDfsLinkStream(Path doc) throws IOException;
```

Comment:
Minor afterthought. For parallelism with "Directory"Stream, this is really a DfsNamespaceStream, not a DfsLinkStream. I don't have a strong preference, and "Link" is shorter, but I thought I'd mention it.